packages/haul-core/src/webpack/plugins/PreloadBundlesPlugin.ts

```diff
@@ -0,0 +1,34 @@
+import webpack from 'webpack';
+
+/**
+ * Preload required bundles for multi-bundle v1 in bundle bootstrapping logic.
+ */
+export class PreloadBundlesPlugin {
+  private bundles: string[];
+
+  /**
+   * Constructs new `PreloadBundlesPlugin`.
+   *
+   * @param config - Config object with `bundles` array. Each value should represent a bundle name.
+   */
+  constructor({ bundles }: { bundles?: string[] }) {
+    this.bundles = bundles || [];
+  }
+
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.compilation.tap('PreloadBundlesPlugin', compilation => {
+      (compilation.mainTemplate as any).hooks.bootstrap.tap(
+        'PreloadBundlesPlugin',
+        (source: string) => {
+          const preload = this.bundles.length
+            ? `${this.bundles.map(
+                bundleName =>
+                  `if (!this["${bundleName}"]) { this.bundleRegistryLoad("${bundleName}", true, true); }\n`
+              )}\n`
```

Comment:
if we have `dedent` in deps somewhere, then it could be used to make it easier to read the template, but that's just a suggestion